### PR TITLE
xz: update to 5.6.1

### PR DESCRIPTION
**Pull request by srirangav**
#### Description

Update xz to 5.6.1

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 13.6.5 22G621 arm64
Xcode 14.3.1 14E300c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: archivers/xz/Portfile [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
 
-name            xz
-version         5.4.6
-checksums       rmd160  b520491a5cae36c5295859d51eee5573a5953a1f \
-                sha256  913851b274e8e1d31781ec949f1c23e8dbcf0ecf6e73a2436dc21769dd3e6f49 \
-                size    2181925
+github.setup    tukaani-project xz 5.6.1 v
+github.tarball_from releases
+revision        0
+
```
Comment (ryandesign) [inline archivers/xz/Portfile:10]:
```suggestion
```

File: archivers/xz/Portfile [hunk 2]
```diff
@@ -22,7 +25,6 @@ XZ utils consist of a few relatively separate submodules: \
 \n* Scripts to ease grepping, diffing and viewing (lz*grep, lzdiff/lzcmp, lzmore/lzless)
 
 homepage        https://tukaani.org/xz/
```
Comment (ryandesign) [inline archivers/xz/Portfile:27]:
```suggestion
homepage        https://tukaani.org/xz/
github.tarball_from \
                releases
```

**Review by ryandesign — CHANGES_REQUESTED** [review]
Thanks. `revision` is set in the subports, and since `github.tarball_from` is the github portgroup's equivalent of `master_sites` I prefer to place it at the location in the Portfile where `master_sites` typically goes (after `homepage`).

File: archivers/xz/Portfile (hunk 2 above)
Comment (srirangav) [reply archivers/xz/Portfile:25]: Hi @ryandesign, I have made these changes to the Portfile and have updated my commit. Thanks.

**Comment by l2dy** [comment]
A bad timing. https://www.openwall.com/lists/oss-security/2024/03/29/4 "backdoor in upstream xz/liblzma"

We should revert to 5.4.x until things settle.

**Comment by pmetzger** [comment]
It looks from subsequent commits that yet more little "tweaks" appear to have been made by the offending party, so we should stay on an old version until the honest xz maintainers feel they're in a more solid position.

**Pull request merged**